core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+        """Helper function that recursively traverses an interaction
+        customization argument spec to locate any SubtitledHtml or
+        SubtitledUnicode objects, and applies a conversion function to the
+        customization argument value.
+
+        Args:
+            schema: dict. The customization dict to be modified: dict
+                with a single key, 'value', whose corresponding value is the
+                value of the customization arg.
+            value: dict. The current nested customization argument value to be
+                modified.
+            conversion_fn: function. The function to be used for converting the
+                content. It is passed the customization argument value and
+                obj_type.
+
+        Returns:
+            dict. The converted customization dict.
+        """
+        schema_type = schema['type']
+        schema_obj_type = schema.get('obj_type', None)
+
+        if (schema_obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_UNICODE or
+                schema_obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML):
+            value = conversion_fn(value, schema_obj_type)
+        elif schema_type == schema_utils.SCHEMA_TYPE_LIST:
+            for i in python_utils.RANGE(len(value)):
+                value[i] = (
+                    InteractionCustomizationArg.traverse_by_schema_and_convert(
+                        schema['items'],
+                        value[i],
+                        conversion_fn
+                    )
+                )
+        elif schema_type == schema_utils.SCHEMA_TYPE_DICT:
+            for property_spec in schema['properties']:
+                name = property_spec['name']
+                value[name] = (
+                    InteractionCustomizationArg.traverse_by_schema_and_convert(
+                        property_spec['schema'],
+                        value[name],
+                        conversion_fn)
+                )
+
+        return value
+
+    @staticmethod
+    def traverse_by_schema_and_get(
+            schema, value, search_obj_types, value_extractor):
+        """Recursively traverses an interaction customization argument spec to
+        locate values with obj_type in search_obj_types, and extracting the
+        value using a value_extractor function.
+
+        Args:
+            schema: dict. The customization dict to be modified: dict
+                with a single key, 'value', whose corresponding value is the
+                value of the customization arg.
+            value: dict. The current nested customization argument value to be
+                modified.
+            search_obj_types: tuple|Set|list. The specified obj_types to extract
+                values from.
+            value_extractor: function. The function that extracts the wanted
+                value from each value that matches the obj_type's.
+
+        Returns:
+            list(*). The extracted values of the matched obj_type's.
```

Comment:
Please elaborate on what "value" means. Explain why "*" is the correct element type and why it can't be more specific.